src/types/label.js

```diff
@@ -196,25 +202,28 @@ function getCalloutSideAdjust(position, options) {
   return side;
 }
 
-function resolveCalloutPosition(element, options) {
+function resolveCalloutPosition(element, options, rotation) {
   const position = options.position;
   if (position === 'left' || position === 'right' || position === 'top' || position === 'bottom') {
     return position;
   }
-  return resolveCalloutAutoPosition(element, options);
+  return resolveCalloutAutoPosition(element, options, rotation);
 }
 
-function resolveCalloutAutoPosition(element, options) {
+const positions = ['left', 'bottom', 'top', 'right'];
+
+function resolveCalloutAutoPosition(element, options, rotation) {
```

Comment:
The `element` should be renamed to `properties` as its not the element, but the newly resolved properties to be animated to the element. Because of the variable name, I thought this was using the current element properties, which would be wrong.

Reply:
You' re right. Properties sounds really better. This also will explain why the center is calculated without using the `getCenterPoint`.